src/apps/loan-list/list/ToggleListViewButtons.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { FC, useCallback, useEffect } from "react";
+import { useText } from "../../../core/utils/text";
+import IconList from "../../../components/icon-list/icon-list";
+import IconStack from "../../../components/icon-stack/icon-stack";
+import { ListView } from "../../../core/utils/types/list-view";
+import { getUrlQueryParam } from "../../../core/utils/helpers/url";
+import { isDate } from "../../../core/utils/helpers/date";
+import { getModalIds } from "../../../core/utils/helpers/general";
+import { useModalButtonHandler } from "../../../core/utils/modal";
+import RenewLoansModal from "../modal/renew-loans-modal";
+import { LoanType } from "../../../core/utils/types/loan-type";
+
+export interface ToggleListViewButtonsProps {
+  setView: (view: ListView) => void;
+  view: ListView;
+  disableRenewLoansButton: boolean;
+  pageSize: number;
+  loans: LoanType[];
+}
+
+const ToggleListViewButtons: FC<ToggleListViewButtonsProps> = ({
+  setView,
+  view,
+  disableRenewLoansButton,
+  pageSize,
+  loans
+}) => {
+  const t = useText();
+  const { open } = useModalButtonHandler();
+  const { allLoansId } = getModalIds();
+
+  const setViewHandler = useCallback(
+    (inputView: ListView) => {
+      setView(inputView);
+    },
+    [setView]
+  );
+
+  const openRenewLoansModal = useCallback(() => {
+    open(allLoansId as string);
+  }, [allLoansId, open]);
+
+  useEffect(() => {
+    const modalString = getUrlQueryParam("modal");
+
+    // If the queryparme is a date, the view should be stacked
+    // because the modal it opens can only be opened in the stacked view
+    if (isDate(modalString)) {
+      setView("stack");
+    }
```

Comment:
This seems really funky. I don't have an overview of the functionality but I think there is a code smell here.
Maybe you disagree but this seems to be fragile and could break if it gets extended.
Could it be an alternative to set an url parameter called view?

Reply:
removed